docs/architecture/architecture.md

```diff
@@ -609,6 +614,133 @@ rootfs by calling `mount(8)` within the counter.  If the devicemapper block devi
 is used, `/` will be mounted on `/dev/vda`.  Users can disable direct mounting
 of the underlying block device through the runtime configuration.
 
+## Kubernetes support
+[Kubernetes\*](https://github.com/kubernetes/kubernetes/) is a popular open source
+container orchestration engine. It runs a control plane where a scheduler
+(typically running on a dedicated master node) calls into a compute node container
+agent called `kubelet`. On each compute node a `kubelet` instance is responsible
+for managing the node pods lifecycle and eventually relies on a container runtime for
+handling the container execution parts. The `kubelet` architecture decouples the
+lifecycle management part from the container execution one through the dedicated
+[`gRPC`](https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/apis/cri/v1alpha1/runtime/api.proto) 
+[Container Runtime Interface (CRI)](https://github.com/kubernetes/community/blob/master/contributors/design-proposals/node/container-runtime-interface-v1.md).
+In other words, `kubelet` is a `CRI` client and expects a `CRI` implementation to
+handle the server side of the interface.
+[CRI-O\*](https://github.com/kubernetes-incubator/cri-o) is a `CRI` implementation
+that relies on `OCI` compatible runtimes for managing container instances.
+
+Clear Containers, being an OCI compatible runtime, naturally fits into the `CRI-O`
+architecture and requirements, and is one of the two officially supported `CRI-O` runtimes.
+However, due to the fact that Kubernetes execution units are sets of containers (also
+known as pods) rather than single containers, the Clear Containers runtime needs to
+get extra information to seamlessly integrate with Kubernetes.
+
+### Problem statement
+
+The [Kubernetes\*](https://github.com/kubernetes/kubernetes/) execution unit is a pod,
+i.e. a set of containers sharing the same resources and networking, mount and PID
+namespaces. A pod specification is a set of containers with an associated list of
+constraints (namespaces, cgroups, hardware resources, security contexts, *etc*).
+By default the Clear Containers runtime associates each container with a dedicated
```

Comment:
We are missing the difference between the environments of the paragraph starting on 644 and the one starting on 650 as both begin with "by default". Is 644 is talking about a node using Docker to start pods? Then is 650 about a node using CRI-O? I suggest we re work both paragraphs.

644 to 648:

By default, Kubernetes uses Docker to start pods and containers within a pod. In these nodes, the Clear Container runtime will associate each container with its own dedicated virtual machine. However, implementing one virtual machine per container is very difficult. The task of providing the Kubernetes pod semantics is complex given the resources of these virtual machines (such as networking or PID) need to be shared.

Then for 650 - 652, I think there is a typo on 151 and "Kubernetes" should be "CRI-O". Is this accurate:

For nodes using the CRI-O runtime, the Clear Containers runtime will still create a virtual machine for each container, however, it can also create and launch one virtual machine per pod. Each container within a pod is then simply a container inside the virtual machine boundary.